File: 3871-minimum-deletions-for-at-most-k-distinct-characters/3871-minimum-deletions-for-at-most-k-distinct-characters.go
func minDeletion(s string, k int) int {
    freq := make([]int, 26)
    dis := 0
    for _, ch := range s {
        idx := int(ch - 'a')
        if freq[idx] == 0 {
            dis++
        }
        freq[idx]++
    }
    
    if k >= dis {
        return 0
    }
    
    del := 0
    sort.Ints(freq)

    j := 0
    for i := 0; i < 26 && j < dis - k; i++ {
        if freq[i] != 0 {
            del += freq[i]
            j++
        }
    }
    return del
}

File: 3871-minimum-deletions-for-at-most-k-distinct-characters/3871-minimum-deletions-for-at-most-k-distinct-characters.cs
public class Solution {
    public int MinDeletion(string s, int k) {
        var freq = new int[26];
        var dis = 0;
        
        foreach (var ch in s) {
            int idx = ch - 'a';
            if (freq[idx] == 0)
                dis++;
            freq[idx]++;
        }
        
        if (k >= dis)
            return 0;
        
        Array.Sort(freq);
        int del = 0;
        
        for (int i = 0, j = 0; i < 26 && j < dis - k; i++) {
            if (freq[i] != 0) {
                del += freq[i];
                j++;
            }
        }

        return del;
    }
}